components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -1163,6 +1270,7 @@ static void psa_key_management_operation(void)
                 case PSA_ALLOCATE_KEY: {
                     status = psa_allocate_key(&psa_key_mng.handle);
                     if (status == PSA_SUCCESS) {
+                        psa_crypto_access_control_register_handle(psa_key_mng.handle, partition_id);
```

Comment:
only in cases of register you have to move the permitted check up
and add 
if (psa_key_mng.func == PSA_ALLOCATE_KEY && psa_key_mng.handle != NULL )
psa_crypto_access_control_register_handle

Reply:
no, we call `psa_crypto_access_control_register_handle` only if the call to `psa_allocate_key` is successful. What if `psa_allocate_key` fails, then we'll call unregister? we register/unregister only if the library completed the operation successfully.